src/Orc.Theming/Behaviors/MandatoryFieldBehavior.cs

```diff
@@ -0,0 +1,33 @@
+﻿namespace Orc.Theming
+{
+    using System.Windows;
+    using System.Windows.Documents;
+    using Catel.Windows.Interactivity;
+
+    public class MandatoryFieldBehavior : BehaviorBase<FrameworkElement>
+    {
+        public MandatoryFieldBehavior()
+        {
+        }
+
+        public static void SetIsMandatory(DependencyObject element, bool value)
+        {
+            element.SetValue(IsMandatoryProperty, value);
+        }
+
+        public static bool GetIsMandatory(DependencyObject element)
+        {
+            return (bool)element.GetValue(IsMandatoryProperty);
+        }
+
+        public static readonly DependencyProperty IsMandatoryProperty =
+            DependencyProperty.RegisterAttached("IsMandatory", typeof(bool), typeof(MandatoryFieldBehavior), new PropertyMetadata(false, (sender, e) => OnIsMandatoryChanged(sender, e)));
+
+        private static void OnIsMandatoryChanged(DependencyObject sender, DependencyPropertyChangedEventArgs e)
+        {
+            var associatedObject = (FrameworkElement)sender;
+            var myAdornerLayer = AdornerLayer.GetAdornerLayer(associatedObject);
```

Comment:
It is technically possible there is no adorner layer. Please check for null here